### PR TITLE
Fix minor bugs in AdaptivePlaylist

### PR DESCRIPTION
**Pull request by lkiesow**
This patch fixes a few minor bugs in the adaptive playlists
implementation.

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

File: modules/common/src/main/java/org/opencastproject/mediapackage/AdaptivePlaylist.java [hunk 1]
```diff
@@ -82,15 +81,12 @@
...
-  List<String> extMaster = new ArrayList<String>(
-          Arrays.asList("#EXT-X-MEDIA:", "#EXT-X-STREAM-INF:", "#EXT-X-I-FRAME-STREAM-INF:", "#EXT-X-SESSION-DATA:"));
-  Pattern masterPatt = Pattern.compile(extMaster.stream().collect(Collectors.joining("|")), Pattern.CASE_INSENSITIVE);
-  Pattern variantPatt = Pattern.compile(extVariant.stream().collect(Collectors.joining("|")), Pattern.CASE_INSENSITIVE);
-  Predicate<String> masterTags = f -> masterPatt.matcher(f) != null;
-  Predicate<String> variantTags = f -> variantPatt.matcher(f) != null;
+  List<String> extMaster = Arrays.asList("#EXT-X-MEDIA:", "#EXT-X-STREAM-INF:", "#EXT-X-I-FRAME-STREAM-INF:",
+          "#EXT-X-SESSION-DATA:");
+  Pattern masterPatt = Pattern.compile(String.join("|", extMaster), Pattern.CASE_INSENSITIVE);
+  Pattern variantPatt = Pattern.compile(String.join("|", extVariant), Pattern.CASE_INSENSITIVE);
```
Comment (lkiesow) [inline modules/common/src/main/java/org/opencastproject/mediapackage/AdaptivePlaylist.java:89]: Mostly simplifications, but `masterPatt` and `variantPatt` always evaluate to `true` and make no sense. Luckily, they are also not used.

File: modules/common/src/main/java/org/opencastproject/mediapackage/AdaptivePlaylist.java [hunk 2]
```diff
@@ -115,16 +111,10 @@ static boolean hasHLSPlaylist(Collection<MediaPackageElement> elements) {
...
-    Collections.sort(fmp4, new Comparator<Track>() {
-      @Override
-      public int compare(Track lhs, Track rhs) {
-        // -1 - less than, 1 - greater than, 0 - equal, all inverted for descending
-        return FilenameUtils.getBaseName(lhs.getURI().getPath())
-                .compareTo(FilenameUtils.getBaseName(rhs.getURI().getPath()));
-      }
-    });
+    }
+    fmp4.sort(Comparator.comparing(track -> FilenameUtils.getBaseName(track.getURI().getPath())));
```
Comment (lkiesow) [inline modules/common/src/main/java/org/opencastproject/mediapackage/AdaptivePlaylist.java:117]: Less confusing and custom comparator but does the exact same thing.

File: modules/common/src/main/java/org/opencastproject/mediapackage/AdaptivePlaylist.java [hunk 3]
```diff
@@ -330,15 +320,12 @@ static void hlsRewriteFileReference(File srcFile, Map<String, String> mapNames)
...
-    BufferedReader br = null;
     // Many tags reference URIs - not all are dealt with in this code, eg:
     // "#EXT-X-MAP:", "#EXT-X-MEDIA:", "#EXT-X-I-FRAME-STREAM-INF:", "#EXT-X-SESSION-DATA:", "#EXT-X-KEY:",
     // "#EXT-X-SESSION-DATA:"
-    try {
-      hlsReWriter = new FileWriter(destFile.getAbsoluteFile(), false);
-      br = new BufferedReader(new FileReader(srcFile));
-      String line = null;
+    try (FileWriter hlsReWriter = new FileWriter(destFile.getAbsoluteFile(), false);
+         BufferedReader br = new BufferedReader(new FileReader(srcFile))) {
```
Comment (lkiesow) [inline modules/common/src/main/java/org/opencastproject/mediapackage/AdaptivePlaylist.java:327]: The old implementation could lead to NullPointerExceptions, this one should be safe with less unnecessary code.

**Review by KatrinIhler — APPROVED** [review]
Code looks good. :)

**Pull request merged**